### PR TITLE
Add support for the bad request error details

### PR DESCRIPTION
**Pull request by shuheiktgw**
SSIA. I've added support for the BadRequest error details.

### CODE REVIEW

File: resolver/resolver.go [hunk 1]
```diff
@@ -2365,9 +2395,8 @@ func (r *Resolver) resolveMessageValidationErrorDetailCELValues(ctx *context, en
 						source.MessageValidationDetailPreconditionFailureLocation(msg.File.Name, msg.Name, vIdx, dIdx, fIdx, fvIdx, "type"),
 					),
 				)
-				return
```
Comment (shuheiktgw) [inline resolver/resolver.go:2368]: I've slightly updated the logic so that one failure won't block others

File: source/file.go [hunk 2]
```diff
@@ -988,7 +988,7 @@ func (f *File) nodeInfoByMessageValidation(list []*ast.MessageLiteralNode, valid
 	for _, elem := range literal.Elements {
 		fieldName := elem.Name.Name.AsIdentifier()
 		switch {
-		case validation.Rule && fieldName == "error":
```
Comment (shuheiktgw) [inline source/file.go:991]: This was a mistake

File: resolver/resolver.go [hunk 3]
```diff
@@ -2411,6 +2438,43 @@ func (r *Resolver) resolveMessageValidationErrorDetailCELValues(ctx *context, en
 			}
 		}
 	}
+
+	for bIdx, badRequest := range detail.BadRequests {
+		for fvIdx, validation := range badRequest.FieldViolations {
```
Comment (goccy) [inline resolver/resolver.go:2443]: `s/validation/violation/` ?

Comment (shuheiktgw) [reply resolver/resolver.go]: 🤦

**Review by goccy — APPROVED** [review]
LGTM 👍

**Pull request merged**